### PR TITLE
Analyzer: Test class should have test method

### PR DESCRIPTION
**Pull request by engyebrahim**
fix: https://github.com/microsoft/testfx/issues/2236

### CODE REVIEW

File: src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    context => AnalyzeSymbol(context, testClassAttributeSymbol, testMethodAttributeSymbol, assemblyInitializationAttributeSymbol, assemblyCleanupAttributeSymbol),
+                    SymbolKind.NamedType);
+            }
+        });
+    }
+
+    private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol testClassAttributeSymbol, INamedTypeSymbol? testMethodAttributeSymbol,
+        INamedTypeSymbol? assemblyInitializationAttributeSymbol, INamedTypeSymbol? assemblyCleanupAttributeSymbol)
+    {
+        var namedTypeSymbol = (INamedTypeSymbol)context.Symbol;
```
Comment (MarcoRossignoli) [inline src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs]: can you give to the variable a better name, is it the symbol referred to a type(class) or method?

File: src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        bool isTestClass = false;
+        foreach (var classAttribute in namedTypeSymbol.GetAttributes())
+        {
+            if (classAttribute.AttributeClass.Inherits(testClassAttributeSymbol))
+            {
+                isTestClass = true;
```
Comment (MarcoRossignoli) [inline src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs:63]: can we exit if true?

Comment (engyebrahim) [reply src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs:63]: I need to check the test method as well

File: src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs (hunk 1 above)
Comment (engyebrahim) [reply src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs]: we are using it like that for class in all analyzers

File: src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+                if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, assemblyInitializationAttributeSymbol)
+                    || SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, assemblyCleanupAttributeSymbol))
+                {
+                    hasAssemblyAttribute = true;
+                }
+            }
+        }
+
+        if (!hasTestMethod && (!namedTypeSymbol.IsStatic
```
Comment (MarcoRossignoli) [inline src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs]: can you put the condition inside the parenthesis in the same line for readability?

File: src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs (hunk 2 above)
Comment (engyebrahim) [reply src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs:63]: oh yup we can I thought it's the other forech

File: src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs]: ok so to me it's not clear in all analyzers at this line I have to go up and "elaborate" the `RegisterCompilationStartAction`

Comment (engyebrahim) [reply src/Analyzers/MSTest.Analyzers/TestClassShouldHaveTestMethodAnalyzer.cs]: ok I'll make it classSymbol

File: test/UnitTests/MSTest.Analyzers.UnitTests/TestClassShouldHaveTestMethodAnalyzerTests.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            """;
+
+        await VerifyCS.VerifyAnalyzerAsync(
+            code,
+            VerifyCS.Diagnostic(TestClassShouldHaveTestMethodAnalyzer.TestClassShouldHaveTestMethodRule)
+                .WithLocation(0)
+                .WithArguments("MyTestClass"));
+    }
+
+    public async Task WhenStaticTestClassWithoutAssemblyAttributes_DoesNotHaveTestMethod_Diagnostic()
```
Comment (MarcoRossignoli) [inline test/UnitTests/MSTest.Analyzers.UnitTests/TestClassShouldHaveTestMethodAnalyzerTests.cs:88]: what if

```cs
 [TestClass]
public class Base 
{
   protected void Helper(){..}
 }

public class Derived : Base 
{
 [TestMethod]
 public void MyTestMethod()
}
```
or

```cs
 [TestClass]
public abstract class Base 
{
   protected void Helper(){..}
 }

public class Derived : Base 
{
 [TestMethod]
 public void MyTestMethod()
}
```

**Review by MarcoRossignoli — APPROVED** [review]

File: test/UnitTests/MSTest.Analyzers.UnitTests/TestClassShouldHaveTestMethodAnalyzerTests.cs (hunk 4 above)
Comment (engyebrahim) [reply test/UnitTests/MSTest.Analyzers.UnitTests/TestClassShouldHaveTestMethodAnalyzerTests.cs:88]: these are not supported in mstest

**Pull request merged**